lib/files.js

```diff
@@ -0,0 +1,54 @@
+const { checkIfPathMatches } = require('./parseOwners')
+
+/**
+ * process the PR based on dependent settings
+ *
+ * @return
+ *  JSON object with the properties mergeable and the description if
+ *  `mergeable` is false
+ */
+module.exports = async (pr, context, settings) => {
+  let fileSettings = settings.files
+  if (!fileSettings) {
+    return {
+      mergeable: true,
+      description: 'Okay To Merge'
+    }
+  }
+
+  let patternToMatch = settings.files.pattern
+  // fetch the file list
+  let result = await context.github.pullRequests.getFiles(context.repo({number: pr.number}))
+  let addedFiles = result.data.filter(file => file.status === 'added')
+  // parse the files to match the pattern
+  const filesToCheck = addedFiles.filter(file => checkIfPathMatches(file.filename, patternToMatch))
+
+  // fetch the file content
+  result = await fetchFileContents(filesToCheck, context)
+  const failedFiles = result.filter(file => {
+    const content = Buffer.from(file.data.content, 'base64').toString()
+
+    return !content.includes(settings.files.header)
```

Comment:
If I'm not mistaken this only checks if the content exists in the string and not necessarily from line position 0? Perhaps `startsWith` will be more appropriate here?

Reply:
Also probably want to add a test where "Test Header" is in the middle of the content of a file. And it should return false...

Reply:
done